android-kotlin/app/src/main/java/com/mongodb/app/ItemsActivity.kt

```diff
@@ -1,39 +1,154 @@
+            val partition = realmApp.currentUser().toString()
+
+            // Initialize a connection to a realm containing all of the Items in this project
+            val config = SyncConfiguration.Builder(user, partition).build()
+
+            // Sync all realm changes via a new instance, and when that instance has been successfully created connect it to an on-screen list (a recycler view)
+            Realm.getInstanceAsync(config, object : Realm.Callback() {
+                override fun onSuccess(realm: Realm) {
+                    // since this realm should live exactly as long as this activity, assign the realm to a member variable
+                    this@ItemsActivity.userRealm = realm
+                    setUpRecyclerView(realm, realmApp.currentUser(), partition)
+                }
+            })
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_items)
         if (realmApp.currentUser() == null) {
             startActivity(Intent(this, LoginActivity::class.java))
         }
-        logoutButton = findViewById(R.id.button_log_out)
-        logoutButton.isEnabled = true
-        logoutButton.setOnClickListener{ (logOut()) }
-    }
-
-    private fun logOut() {
-        // while this operation completes, disable the button to logout
-        logoutButton.isEnabled = false
-        realmApp.currentUser()?.logOutAsync {
-            // re-enable the button after user registration returns a result
-            logoutButton.isEnabled = true
-            if (it.isSuccess) {
-                realmApp.removeUser(realmApp.currentUser())
-                Log.v(TAG(), "user logged out")
-                startActivity(Intent(this, LoginActivity::class.java))
-            } else {
-                Log.e(TAG(), "log out failed! Error: ${it.error}")
+
+        fab = findViewById(R.id.floating_action_button)
+        fab.isEnabled = true
+        fab.setOnClickListener { (addItem()) }
+
+        recyclerView = findViewById(R.id.items_list)
+
+        val toolbar = findViewById<View>(R.id.task_menu) as Toolbar
+        setSupportActionBar(toolbar)
+        toolbar.showOverflowMenu()
+    }
+
+    // Ensure the user realm closes when the activity ends.
+    override fun onDestroy() {
+        super.onDestroy()
+        userRealm?.close()
+        recyclerView.adapter = null
+    }
+    override fun onStop() {
+        super.onStop()
+        realmApp.currentUser().run {
+            userRealm?.close()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.task_menu, menu)
+        return true
+    }
+
+    /**
+     *  Logs out the user and brings them back to the login screen.
+     */
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val user = realmApp.currentUser()
```

Comment:
This is declared well before needed.

Reply:
you're right, I removed it from being a local variable in this method at all